lib/Serialization/SerializedModuleLoader.cpp

```diff
@@ -142,6 +142,23 @@ bool SerializedModuleLoader::maybeDiagnoseArchitectureMismatch(
   return true;
 }
 
+static std::pair<llvm::SmallString<16>, llvm::SmallString<16>>
+getArchSpecificModuleFileNames(StringRef archName) {
```

Comment:
Why not pass along the Triple rather than StringRef?  It feels like it would be nicer to derive that from the triple rather than the architecture name specifically.  We could have a boolean flag for whether it should do the fallback or not.

Reply:
I don't think that actually makes the call sites nicer, and as a helper function that's the important thing.